### PR TITLE
fix: correct virial sign, pressure mode forwarding, barostat mass, and cell KE in NPT/NPH integrators

### PR DESCRIPTION
**Pull request by zubatyuk**
# ALCHEMI Toolkit Pull Request

## Description

Fix multiple bugs in the NPT and NPH integrators related to virial sign conventions, pressure coupling mode forwarding, barostat mass scaling, and cell kinetic energy computation.

## Type of Change

- [x] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Performance improvement
- [ ] Documentation update
- [x] Refactoring (no functional changes)
- [ ] CI/CD or infrastructure change

## Related Issues

Companion to NVIDIA/nvalchemi-toolkit-ops#66 which fixed the ops-layer strain rate, cell kinetic energy, and virial sign bugs. This PR applies the corresponding toolkit-side fixes: forwarding the `pressure_coupling` mode string and `cells_inv` through the bridge, correcting the virial sign convention at the bridge boundary, fixing barostat mass scaling and NHC DOF counts for non-isotropic modes, and aligning cell kinetic energy with the corrected ops kernel.

### Bug 1 — Virial sign convention (tension-positive bridge)

The `compute_pressure_tensor` bridge wrapper now accepts stress in the tension-positive convention (matching `batch.stress` and ASE) and internally negates it (`virial = -stress`) before passing to the ops kernel which expects compression-positive virial. Previously the raw stress was passed through without negation, producing pressure with the wrong sign.

### Bug 2 — Pressure coupling mode not forwarded to ops kernels

- `npt_velocity_half_step` and `nph_velocity_half_step` now accept a `pressure_mode` string parameter (defaulting to `"isotropic"`) and pass it directly to the underlying ops drag kernels. Previously only isotropic drag was ever applied regardless of the `pressure_coupling` setting. The integrators pass `self.pressure_coupling` through without any intermediate encoding.
- `npt_barostat_half_step` and `nph_barostat_half_step` now use `_target_pressure_wp_array` to correctly convert scalar `[M]`, anisotropic `[M,3]`, or triclinic `[M,3,3]` target pressure tensors to the matching Warp array dtype. Previously only scalar was supported.
- `_to_per_system` in `_bridge.py` was generalized to preserve trailing dimensions so that anisotropic and triclinic pressure tensors are broadcast correctly.

### Bug 3 — Barostat mass not divided by 3 for non-isotropic modes

In both NPT and NPH `_init_state` and `_make_new_state`, the barostat inertia W (computed by the ops kernel as `(N_f + d) * kT * tau_P^2`) is now divided by 3 for non-isotropic modes. The ops kernel computes W for the full 3x3 cell; for anisotropic/triclinic modes each diagonal (or individual) DOF should carry W/3.

### Cell kinetic energy: ḣ → ε̇

Cell kinetic energy was computed from `||ḣ||^2` (raw cell velocities) instead of `||ε̇||^2` (strain rate `ε̇ = ḣ h⁻¹`). A new `_cell_kinetic_energy` helper computes the strain rate and takes the mode-aware Frobenius norm (diagonal only for isotropic/anisotropic, full matrix for triclinic). This aligns the toolkit with the ops `compute_cell_kinetic_energy` kernel.

### Barostat NHC DOF count

The barostat thermostat NHC now uses the correct number of dummy particles: 1 (3 DOFs) for isotropic/anisotropic modes and 3 (9 DOFs) for triclinic. Previously 3 dummy particles (9 DOFs) were used unconditionally.

## Testing

- [x] Unit tests pass locally (`make pytest`)
- [x] Linting passes (`make lint`)
- [x] New tests added for new functionality meets coverage expectations?

New tests added:
- `test_npt_nph.py`: pressure shape preservation for anisotropic/triclinic, NHC barostat chain mass shape for all modes, W/3 verification for non-isotropic modes, pre/post_update smoke tests for anisotropic and triclinic modes (both NPT and NPH).
- `test_ops.py`: `test_compute_pressure_tensor_tension_positive_convention` verifying that positive (tensile) stress yields negative pressure through the bridge. Fixed `volumes` initialization from `torch.zeros` to `torch.full(..., 2.0)` to avoid division by zero.

## Checklist

- [x] I have read and understand the [Contributing Guidelines](../CONTRIBUTING.md)
- [ ] I have updated the [CHANGELOG.md](../CHANGELOG.md)
- [x] I have performed a self-review of my code
- [x] I have added docstrings to new functions/classes
- [ ] I have updated the documentation (if applicable)

## Additional Notes

All six modified files and their roles:
- `_bridge.py`: generalized `_to_per_system` to preserve trailing dimensions for anisotropic/triclinic pressure tensors
- `npt_nph.py`: `_target_pressure_wp_array` helper, virial negation in `compute_pressure_tensor`, `pressure_mode: str` parameter on velocity half-step functions
- `npt.py`: `_cell_dof_count`, `_cell_kinetic_energy` helpers, W/3 scaling, mode forwarding, NHC DOF fix
- `nph.py`: W/3 scaling, mode forwarding
- `test_npt_nph.py`: 10 new test cases covering non-isotropic modes for both NPT and NPH
- `test_ops.py`: 1 new test case for virial sign convention, volumes fix in helper

> [!TIP]
> This repository uses Greptile, an AI code review service, to help conduct
> pull request reviews. We encourage contributors to read and consider suggestions
> made by Greptile, but note that human maintainers will provide the necessary
> reviews for merging: Greptile's comments are **not** a qualitative judgement
> of your code, nor is it an indication that the PR will be accepted/rejected.
> We encourage the use of emoji reactions to Greptile comments, depending on
> their usefulness and accuracy.

### CODE REVIEW

File: nvalchemi/dynamics/_ops/_bridge.py [hunk 1]
```diff
@@ -84,13 +85,52 @@ def _to_per_system(
...
-        return val.expand(M).to(device=device, dtype=dtype).contiguous()
+        if val.ndim == 0:
+            return val.expand(M).contiguous()
+        leading = val.shape[0]
+        if leading not in (1, M):
+            raise ValueError(
+                f"Expected leading dimension 1 or {M} for per-system broadcast, "
+                f"got shape {tuple(val.shape)}."
+            )
+        return val.expand((M, *val.shape[1:])).contiguous()
```
[the remark this answers is not in the document]
Comment (zubatyuk) [reply nvalchemi/dynamics/_ops/_bridge.py]: Fixed in 2e388bfe19aea21d30f7c6e7f7225bd4199399bf.

**Review by dallasfoster — APPROVED** [review]
